core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -43,15 +46,24 @@ public static MetricsConfig getDefault() {
 
   public static MetricsConfig fromProperties(Map<String, String> props) {
     MetricsConfig spec = new MetricsConfig();
+    String defaultModeAsString = props.getOrDefault(DEFAULT_WRITE_METRICS_MODE, DEFAULT_WRITE_METRICS_MODE_DEFAULT);
+    spec.defaultMode = MetricsModes.fromString(defaultModeAsString);
```

Comment:
Will we fail jobs if the default mode is invalid?
Will it make sense to fallback `DEFAULT_WRITE_METRICS_MODE_DEFAULT`?

Reply:
Good idea, we should wrap that in a try/catch as well.

Reply:
Fixed.